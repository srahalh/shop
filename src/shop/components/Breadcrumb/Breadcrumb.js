import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { useGetProductByIdQuery } from '../../redux/API/productsAPI';

import "./Breadcrumb.scss";

const Breadcrumb = () => {
  const navigate = useNavigate();
  const { pathname } = useLocation();
  const { data = [], isLoading } = useGetProductByIdQuery(
    pathname?.split("/")[2]
  );

  return (
    <ul className="breadcrumb__list">
      <li>
        <a onClick={() => navigate("/")}>Products</a>
      </li>
      {pathname.split("/")[1] === "product" && !isLoading && (
        <li>
          <a>{data?.title}</a>
        </li>
      )}
    </ul>
  );
};

export default Breadcrumb;
