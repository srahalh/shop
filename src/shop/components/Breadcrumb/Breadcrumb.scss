@import '../../styles/config.scss';

.breadcrumb__list {
  padding: $padding-md 0;
  list-style: none;
  align-self: flex-start;
  margin: 0;

  li {
    display: inline;
    font-size: $font-sm;
    color: $color-primary;

    a {
      color: $color-secondary;
      text-decoration: none;

      &:hover {
        color: darken($color-secondary, 10%);
        text-decoration: underline;
        cursor: pointer;
      }
    }
  }

  li+li::before {
    content: "/\00a0";
    padding: $padding-sm;
  }
}