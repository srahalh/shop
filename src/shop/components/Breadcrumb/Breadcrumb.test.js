import React from "react";
import { renderWithProviders } from "../../test/setupTests";
import Breadcrumb from "./Breadcrumb";
const fulfilledState = require("../../test/fulfilledProductDetailState.json");


describe("Test Breadcrumb", () => {
  test("Should render Breadcrumb and have title of state in snapshot", () => {
    const { asFragment  } = renderWithProviders(<Breadcrumb />, { preloadedState: fulfilledState, route: "/product/9"});
    expect(asFragment ()).toMatchSnapshot();
  });
});
