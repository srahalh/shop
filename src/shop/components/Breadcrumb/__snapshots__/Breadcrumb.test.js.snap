// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Breadcrumb Should render Breadcrumb and have title of state in snapshot 1`] = `
<DocumentFragment>
  <ul
    class="breadcrumb__list"
  >
    <li>
      <a>
        Products
      </a>
    </li>
    <li>
      <a>
        Liquid 5
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
